rag_experiment_accelerator/run/index.py

```diff
@@ -128,26 +127,31 @@ def convert_docs_to_vector_db_records(docs):
     return dicts
 
 
-def embed_chunks(config: IndexConfig, pre_process, chunks):
+def embed_chunks(config: Config, index_config: IndexConfig, pre_process, chunks):
     """
     Generates embeddings for chunks of documents.
 
     Args:
         config (object): A configuration object that holds various settings.
         pre_process (object): An object with a method for preprocessing text.
-        chunks (list): A list of all documents chunks to be embeded.
+        chunks (list): A list of all documents chunks to be embedded.
 
     Returns:
         tuple: A tuple containing the index name and the list of processed documents.
     """
-    with TimeTook(f"generate embeddings for {config.index_name()} ", logger=logger):
+    with TimeTook(
+        f"generate embeddings for {index_config.index_name()} ", logger=logger
+    ):
         embedded_chunks = []
         with ExitStack() as stack:
             executor = stack.enter_context(ThreadPoolExecutor())
 
             futures = {
                 executor.submit(
-                    embed_chunk, pre_process, config.embedding_model, doc
+                    embed_chunk,
+                    pre_process,
+                    config.get_embedding_model(index_config.embedding_model.model_name),
```

Comment:
Same as another embedding model shenanigan with Time

Reply:
Downloading happens once - on config creation (method is create_embedding_model). And this method is just getting it from a dictionary of models.